src/NLog/Targets/IJsonSerializer.cs

```diff
@@ -46,7 +46,19 @@ public interface IJsonSerializer
         /// into JSON format.
         /// </summary>
         /// <param name="value">The object to serialize to JSON.</param>
-        /// <returns>Serialized value.</returns>
+        /// <returns>Serialized value (null = Serialize failed).</returns>
         string SerializeObject(object value);
     }
+
+    internal interface IJsonSerializerV2
```

Comment:
Yes it will not be super optimal for Json.Net as the implementation would be:

```
bool SerializeObject(object value, System.Text.StringBuilder builder)
{
    try
    {
       StringWriter sw = new StringWriter(sb);
       JsonSerializer serializer = new JsonSerializer();
       using (JsonWriter jsonTextWriter = new JsonTextWriter(sw))
       {
            serializer.Serialize(jsonTextWriter, obj);
       }
       return true;
   }
   catch
   {
      return false;
   }
}
```